README.rst

```diff
@@ -8,45 +8,69 @@ and the form of `p(y|x)` can be arbitrarily complex (e.g multimodal, heteroskeda
 
+Installation
+============
+
+You can install Treeffuser via pip from PyPI with the following command::
+
+    pip install treeffuser
+
+You can also install the in-development version with::
+
+    pip install git+https://github.com/blei-lab/tree-diffuser.git@main
+
+
 Usage Example
--------------
+============
+
+Here's a simple example demonstrating the usage of Treeffuser.
 
-Here's how you can use Treeffuser in your project:
+We generate an heteroscedastic response with a sinusoidal mean and fat tails.
 
 .. code-block:: python
 
-    from treeffuser import Treeffuser
+    import matplotlib.pyplot as plt
     import numpy as np
+    from treeffuser import Treeffuser
 
-    # (n_training, n_features), (n_training, n_targets)
-    X, y = ...  # load your data
-    # (n_test, n_features)
-    X_test = ...  # load your test data
-
-    # Estimate p(y|x) with a tree-based diffusion model
-    model = Treeffuser()
-    model.fit(X, y)
+    # Generate data
+    rng = np.random.default_rng(seed=0)
+    n = 5000
+    x = rng.uniform(0, 2 * np.pi, size=n)
+    y = np.sin(x) + rng.laplace(scale=x / 20, size=n)
 
-    # Draw samples y ~ p(y|x) for each test point
-    # (n_samples, n_test, n_targets)
-    y_samples = model.sample(X_test, n_samples=1000)
+We fit Treeffuser and generate samples. These can be used to compute any downstream estimates of interest.
 
-    # Compute downstream metrics
-    mean = np.mean(y_samples, axis=0)
-    std = np.std(y_samples, axis=0)
-    median = np.median(y_samples, axis=0)
-    quantile = np.quantile(y_samples, q=0 axis=0)
-    ... # other metrics
+.. code-block:: python
 
-Please refer to the docstrings for more information on the available methods and parameters.
+    # Fit the model
+    model = Treeffuser()
+    model.fit(x, y)
 
-Installation
-============
+    # Generate samples and return downstream estimates
+    x_new = np.linspace(x.min(), x.max(), 200)
+    y_samples = model.sample(x_new, n_samples=10**2, verbose=True)
+    y_preds = y_samples.mean(axis=0)
+    y_q05, y_q95 = np.quantile(y_samples, q=[0.05, 0.95], axis=0)
+    ... # other metrics
 
-You can install Treeffuser via pip from PyPI with the following command::
+We then plot the original data along with the model's predictions.
 
-    pip install treeffuser
+.. code-block:: python
 
-You can also install the in-development version with::
+    sorted_idx = np.argsort(x_new)
+    x_sorted, y_preds_sorted, y_q05_sorted, y_q95_sorted = [
+        arr[sorted_idx] for arr in [x_new, y_preds, y_q05, y_q95]
+    ]
+    plt.plot(x_sorted, y_preds_sorted, color="black")
+    plt.fill_between(x_sorted, y_q05_sorted, y_q95_sorted, color="gray", alpha=0.4)
+    plt.scatter(x, y, s=1)
+
+    plt.xlabel("x")
+    plt.ylabel("y")
+    plt.show()
+.. image:: README_example.png
+   :alt: Treeffuser on heteroscedastic data with sinuisodal response and fat tails
+   :align: center
```

Comment:
I think the beginning is really great. But starting here, the code gets complicated. Also I am not a fan of the plot (it's jiggly). 

Can the plot show something else? Maybe a smooth ground truth and treeffuser samples? like our figure 1.
To further simplify the code, we could have an example.py inside treeffuser such that the readme becomes 
```python
from treeffuser.examples import simulation_sin_get_data, simulation_sin_plot

x, y, x_test = simulation_sin_get_data(n=1000) 
...
# plotting section
plt.scatter(x, y, s=1)
simulation_data_plot(mean=True, quantiles=[0.05, 0.095])
```

What do you think?

Reply:
After the plot, we can still have a last section saying: 
```python
# Use samples to estimate any quantity of interest from `p(y|x)`
y_mean = y_samples.mean(axis=0)
y_q05, y_q95 = np.quantile(y_samples, q=[0.05, 0.95], axis=0)
y_std = ...
```

Reply:
My point being that we don't have to show those estimates (since they are ugly).
(If really you want to show the predicted mean, then I would suggest to use scatter (no line) and reduce the number of test point to maybe 20, so that it does not appear jiggly. 
But I still think ground truth is more important, to show we are learning the correct thing,

Reply:
good point! i wanted to show probabilistic prediction in action using treeffuser (something more than plotting samples or histograms), but i see how the jiggly plot may be misinterpreted. let's add two components as originally suggested by @velezbeltran and just plot samples vs raw data. i'll add that either tonight or tomorrow.

Reply:
that should greatly simplify the code for plotting. if we want to add something like Figure 1 to show the bimodality of the distribution, we can defer the code to a script as you suggested. i like the idea of closing the example with the "samples are for you to play with" part.